mkosi/__init__.py

```diff
@@ -4244,16 +4246,45 @@ def run_coredumpctl(args: Args, config: Config) -> None:
 def run_serve(args: Args, config: Config) -> None:
     """Serve the output directory via a tiny HTTP server"""
 
-    run(
-        [python_binary(config), "-m", "http.server", "8081"],
-        stdin=sys.stdin,
-        stdout=sys.stdout,
-        sandbox=config.sandbox(
-            network=True,
-            relaxed=True,
-            options=["--chdir", config.output_dir_or_cwd()],
-        ),
-    )
+    with contextlib.ExitStack() as stack:
+        want_storagetm = config.output_format == OutputFormat.disk and config.find_binary(
+            "/usr/lib/systemd/systemd-storagetm"
+        )
+
+        http = stack.enter_context(
+            spawn(
+                [python_binary(config), "-m", "http.server", "8081"],
+                stdin=sys.stdin,
+                stdout=sys.stdout,
+                sandbox=config.sandbox(
+                    network=True,
+                    relaxed=True,
+                    options=["--chdir", config.output_dir_or_cwd()],
+                ),
+                foreground=not want_storagetm,
+            )
+        )
+
+        if want_storagetm:
+            storagetm = stack.enter_context(
+                spawn(
+                    ["/usr/lib/systemd/systemd-storagetm", config.output_with_format],
+                    stdin=sys.stdin,
+                    stdout=sys.stdout,
+                    sandbox=config.sandbox(
+                        network=True,
+                        relaxed=True,
+                        options=["--chdir", config.output_dir_or_cwd()],
+                        setup=become_root_cmd(),
+                    ),
+                    foreground=True,
+                )
+            )
+
+            storagetm.wait()
+            http.send_signal(signal.SIGINT)
+        else:
+            http.wait()
```

Comment:
http gets sent `SIGINT` in the if-branch, but everything gets torn right afterwards, would this maybe make more sense
```suggestion

        http.wait()
```

Reply:
We always wait for the process to finish in spawn() unless an exception is thrown, so no need to wait in this case, though it doesn't hurt either